text/1006-groupped-cli.rst

```diff
@@ -0,0 +1,240 @@
+differently from all other commands is that they are essentially "tools within
+a tool". Technically this is a valid argument: both command groups do not work
+with a single database or even take database connection arguments, contrary
+to most other commands. But in reality, we cannot expect users to understand
+the "tool within a tool" concept without reading RFCs or the documentation.
+Most likely the CLI would be just perceived as inconsistent at best, or
+inconvinient in the worst case.
+
+With adding the ``edgedb project`` subcommand, it became clear that maintaining
+two entirely different naming schemes in one tool will be confusing to the
+users.  Here's what a portion of ``--help`` output would look like if we keep
+the current approach::
+
+    <...>
+
+    Migrations:
+        create-migration         Create a migration script
+        apply-migrations         Bring current database to the latest or
+                                 a specified revision
+        show-migrations          Show all migration versions
+        check-migrations         Show current migration state
+
+    Server Management:
+        server status            Status of an instance
+        server init              Initialize a new server instance
+        server start             Start an instance
+        server stop              Stop an instance
+        server restart           Restart an instance
+        server info              Show server information
+
+    <...>
+
+The actual ``--help`` output will be considerably longer than that. It will
+likely take a few days for the users to not be confused whether to type
+``edgedb project init`` or ``edgedb init-project``, because they have just
+run ``edgedb create-migration``.
+
+Lastly, we will inevitably continue evolving the EdgeDB CLI by adding new
+subcommands. For example, we might add the ``edgedb cloud`` command group which
+would classify as a "tool within a tool". Or we might add subcommands to manage
+authentication and other aspects of database management, which are unlikely
+to be classified as "tools within a tool", in which case we will have more
+commands in the form of ``edgedb create-X`` or ``edgedb show-Y``. Ultimately,
+the number of commands with entirely different naming will only continue to
+grow.
+
+
+Overview
+========
+
+The RFC proposes to have the following comamnds structure for the
+the EdgeDB RC1 release::
+
+    dump                       Create a database backup
+    restore                    Restore a database backup from file
+    configure                  Configure a DB or a server instance
+
+    migration apply            Migrate the database to the latest revision
+    migration create           Create a new migration
+    migration log              Show the migrations log
+    migration status           Show the current migration state
+    migrate                    An alias for `edgedb migration apply`
```

Comment:
This "convenient" alias will make completion uglier (you always have to type `m<tab>i<tab>`

Reply:
> (Is this already implemented? I don't have autocompletion on my machine.)

You may need to run the gen_completions command first: `edgedb _gen_completions --help`

Reply:
They should be generated on installation. What's your shell?

Reply:
> worth introducing a highly visible breaking change 

We are breaking almost every other command in a very visible way. So keeping just this one for the sake of compatibility sounds like a far-fetched argument. Other than that, I don't know if it worth it.